engine/src/main/java/io/camunda/zeebe/engine/state/query/StateQueryService.java

```diff
@@ -64,7 +65,9 @@ private void ensureServiceIsOpened() {
     }
     if (state == null) {
       // service is used for the first time, create state now
-      state = new ZeebeDbState(zeebeDb, zeebeDb.createContext());
+      // we don't need a key generator here, so we set it to null
+      state =
+          new ZeebeDbState(Protocol.DEPLOYMENT_PARTITION, zeebeDb, zeebeDb.createContext(), null);
```

Comment:
:wrench: We could prevent NPE by providing a null object for the key generator.

For example, implementing the key generator interface but returning `-1` as the key or throwing an exception.

Currently, it doesn't hurt to pass `null` as the argument but in general, it could be considered as a code smell.

Reply:
Totally agree.